x-pack/plugins/files/public/components/upload_file/upload_state.ts

```diff
@@ -169,7 +169,8 @@ export class UploadState {
 
     file$.setState({ status: 'uploading', error: undefined });
 
-    const { name, mime } = parseFileName(file.name);
+    const { name } = parseFileName(file.name);
+    const mime = file.type || undefined;
```

Comment:
nit: we could also try extra hard to figure out the mime type (not sure if it is worth it though), like:

```suggestion
    const mime = file.type || mime.lookup(fileName);
```

Reply:
Yeah, I did consider that too, but I think in 99% of cases won't be worth the extra KB.